app/src/main/java/com/github/ybecker/epforuml/database/FirebaseDatabaseAdapter.kt

```diff
@@ -1,59 +1,345 @@
 package com.github.ybecker.epforuml.database
 
 import com.github.ybecker.epforuml.database.Model.*
-import com.google.firebase.database.DatabaseReference
-import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.*
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
 
 /**
  * This class represents a database that uses Firebase Realtime Database
  */
 class FirebaseDatabaseAdapter : Database() {
 
-    private val db: DatabaseReference = FirebaseDatabase.getInstance().reference
+    // save the database reference
+    private val dbInstance = FirebaseDatabase.getInstance("https://epforuml-38150-default-rtdb.europe-west1.firebasedatabase.app")
+    private val db: DatabaseReference = dbInstance.reference
+
+    // save every useful path to navigate in the database
+    private val usersPath = "users"
+    private val coursesPath = "courses"
+    private val questionsPath = "questions"
+    private val answersPath = "answers"
+
+    private val courseIdPath = "courseId"
+    private val userIdPath = "userId"
+    private val questionIdPath = "questionId"
+    private val answerIdPath = "answerId"
+
+    private val courseNamePath = "courseName"
+    private val usernamePath = "username"
+
+    private val questionTextPath = "questionText"
+    private val answerTextPath = "answerText"
 
     override fun availableCourses(): Set<Course> {
```

Comment:
There is some code repetition through the different functions:  getQuestionsForCourse, availableCourses, getQuestionAnswers, getUserQuestions and getUserAnswers have the same structure
Maybe it would be better for exemple to do one single function that recognizes the different paths.